src/main/java/org/wildfly/security/x500/X500.java

```diff
@@ -137,4 +143,55 @@ private X500() {}
             return Arrays.copyOf(certificates, certificates.length, X509Certificate[].class);
         }
     }
+
+    public enum OidCategory {
+        RDN
+    }
+
+    private static Map<OidCategory, Map<String, String>> attributeNameToOidTable = new HashMap<>();
+    private static Map<OidCategory, Map<String, String>> oidToAttributeNameTable = new HashMap<>();
+
+    static { // loading OIDs maps from properties file
+        for (OidCategory category : OidCategory.values()) {
+            attributeNameToOidTable.put(category, new HashMap<>());
+            oidToAttributeNameTable.put(category, new HashMap<>());
+        }
+        try {
+            Properties props = new Properties();
+            props.load(X500.class.getResourceAsStream("oids.properties"));
+            for (String propName : props.stringPropertyNames()) {
+                int delimiter = propName.indexOf('.');
+                if (delimiter < 0) throw new IllegalStateException();
+                OidCategory category = OidCategory.valueOf(propName.substring(0, delimiter));
+                String oid = propName.substring(delimiter + 1);
+                String[] attributeNames = props.getProperty(propName).split(",");
+                for (String attributeName : attributeNames) {
+                    attributeNameToOidTable.get(category).put(attributeName, oid);
+                }
+                oidToAttributeNameTable.get(category).put(oid, attributeNames[0]);
+            }
+        } catch (Exception e) {
+            throw ElytronMessages.log.unableToLoadOidsFromPropertiesFile(e);
+        }
+    }
+
+    /**
+     * Convert an X.500 attribute name (AVA keyword) to OID
+     * @param category category of OID / context of the conversion
+     * @param attributeName X.500 attribute name
+     * @return corresponding OID or {@code null} if was not recognized
+     */
+    public static String attributeNameToOid(OidCategory category, String attributeName) {
+        return attributeNameToOidTable.get(category).get(attributeName.toUpperCase(Locale.ROOT));
```

Comment:
No, because the table is pre-filled by maps for every enum value:
https://github.com/wildfly-security/wildfly-elytron/pull/550/files/8c3901755d610d785895578c85ab8f8077b870da#diff-ab46f06d0dd427d972f60e9c2237ebccR155